flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/AggFunctionFactory.scala

```diff
@@ -366,7 +366,7 @@ class AggFunctionFactory(
         new LeadLagAggFunction.DoubleLeadLagAggFunction(argTypes.length)
       case BOOLEAN =>
         new LeadLagAggFunction.BooleanLeadLagAggFunction(argTypes.length)
-      case VARCHAR =>
+      case VARCHAR | CHAR =>
         new LeadLagAggFunction.StringLeadLagAggFunction(argTypes.length)
```

Comment:
`LeadLagAggFunction.StringLeadLagAggFunction#getResultType` returns `STRING` type, which is a `VARCHAR` type. I think users will not expect feeding a `CHAR` into the lead/lag function and get a `VARCHAR` out. It might be better to add a new type of lead/lag function.

Reply:
Although String is used in these processes, my tests show that the field type returned by Lead and Lag in the result table is still CHAR.As shown below, is this test correct? Thank you.

    val testCardinality = tableEnv.fromValues(
      DataTypes.ROW(
        DataTypes.FIELD("name", DataTypes.STRING()),
        DataTypes.FIELD("email", DataTypes.VARCHAR(20)),
        DataTypes.FIELD("company", DataTypes.CHAR(10))
      ),
      row("Alice", "alice@test.com", "Test Ltd1"),
      row("Alice", "alice@test.com", "Test Ltd2"),
      row("Alice", "alice@test.com", "Test Ltd3")
    )

    tableEnv.executeSql(
      s"""
        |CREATE TEMPORARY VIEW testLeadLag AS
        |SELECT
        |  name,
        |  email,
        |  company,
        |  LEAD(company, 1,'null') over (
        |    partition by name
        |)  AS companyLead,
        |  LAG(company, 1,'null') over (
        |    partition by name
        |)  AS companyLag
        |
        |FROM ${testCardinality}
        |""".stripMargin)
    val tableLeadLag = tableEnv.from("testLeadLag")

    tableLeadLag.printSchema()

result table structure is :

(
  `name` STRING,
  `email` VARCHAR(20),
  `company` CHAR(10),
  `companyLead` CHAR(10) NOT NULL,
  `companyLag` CHAR(10) NOT NULL
)

Reply:
If you dig into the method you'll find out that the result of `Table#printSchema` is from Calcite, not from the Flink SQL runtime. You can also add a breakpoint in `LeadLagAggFunction .StringLeadLagAggFunction#getResultType` and run this test to verify this. It won't pause at the breakpoint.

In short, I still suggest you create a corresponding `LeadLagAggFunction`.

Reply:
Thank you very much. I really didn't know the details of printSchema before.I added a Funciton to handle Char type, which will return a fixed-length Char.Please review it. Thanks again.